lib/kafka/cluster.rb

```diff
@@ -46,6 +46,10 @@ def add_target_topics(topics)
         @target_topics.merge(new_topics)
 
         refresh_metadata!
+
+        cluster_info.topics.each do |topic|
+          Protocol.handle_error(topic.topic_error_code)
+        end
```

Comment:
sure, I can buy that, but presumably some other bit of the code should rescue transient errors and retry?